src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -55,133 +54,147 @@
+    @ApiOperation(value = "Creates the subscriptions")
+    public ResponseEntity<List<SubscriptionResponse>> createSubscription(@RequestBody List<Subscription> subscriptions) {
+        errorMap = new HashMap<>();
+        String user = (authenticate) ? HttpSessionConfig.getCurrentUser() : "";
 
-            String user = (authenticate) ? HttpSessionConfig.getCurrentUser() : "";
-            subscription.setUserName(user);
-            subResponse = null;
+        subscriptions.forEach(subscription -> {
+            String subscriptionName = subscription.getSubscriptionName();
             try {
-                subscription.setCreated(Instant.now().toEpochMilli());
+                LOG.debug("Subscription creation has been started: " + subscriptionName);
                 subscriptionValidator.validateSubscription(subscription);
-            } catch (SubscriptionValidationException e) {
-                String msg = "Validation of Subscription parameters on:" + subscription.getSubscriptionName()
-                        + " failed! Error: " + e.getMessage();
-                LOG.error(msg);
-                subscriptionResponse.setMsg(msg);
-                subscriptionResponse.setStatusCode(HttpStatus.PRECONDITION_FAILED.value());
-                return new ResponseEntity<>(subscriptionResponse, HttpStatus.PRECONDITION_FAILED);
-            }
 
-            if (!subscriptionService.doSubscriptionExist(subscription.getSubscriptionName())) {
-                subscriptionService.addSubscription(subscription);
-                LOG.info("Subscription :" + subscription.getSubscriptionName() + " Inserted Successfully");
-                subscriptionResponse.setMsg("Inserted Successfully");
-                subscriptionResponse.setStatusCode(HttpStatus.OK.value());
-                subResponse = new ResponseEntity<>(subscriptionResponse, HttpStatus.OK);
-
-            } else {
-                LOG.error("Subscription :" + subscription.getSubscriptionName() + " already exists");
-                subscriptionResponse.setMsg("Subscription already exists");
-                subscriptionResponse.setStatusCode(HttpStatus.BAD_REQUEST.value());
-                subResponse = new ResponseEntity<>(subscriptionResponse, HttpStatus.BAD_REQUEST);
+                if (!subscriptionService.doSubscriptionExist(subscriptionName)) {
+                    subscription.setUserName(user);
+                    subscription.setCreated(Instant.now().toEpochMilli());
+                    subscriptionService.addSubscription(subscription);
+                    LOG.debug("Subscription is inserted successfully: " + subscriptionName);
+                } else {
+                    LOG.error("Subscription already exists: " + subscriptionName);
+                    errorMap.put(subscriptionName, "Subscription already exists");
+                }
+            } catch (Exception e) {
+                LOG.error("Error on subscription " + subscriptionName + ". " + e.getMessage());
```

Comment:
Since this is a the Create Subscription error, so the message can be more clear/suitable in the log for this specific situation:
"Failed to create subscription " + subscriptionName + "\nError Message: " + e.getMessage()

Reply:
yes. I will add